cmd/helm/list.go

```diff
@@ -66,7 +66,6 @@ func newListCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 			if client.AllNamespaces {
 				client.SetConfiguration(newActionConfig(true))
 			}
-			client.All = client.Limit == -1
```

Comment:
it looks like this got removed but nothing was added back to restore this functionality. Was this intentional, and if so is there a way we can add this back in?

Reply:
Hi @bacongobbler !
I almost forgot why i removed this :)

This was intentionally done in this commit https://github.com/helm/helm/pull/5539/commits/f4c2b02cef95aecad6d691b812e4508fefb7ac6b. When i was fixing actual issue with the flags for #5499 i also found issues with `--all` and `--superseded` flags, so as part of fixing them i removed this line.
Also, the test cases were buggy for these ^ two flags so CI didn't complain so far about them, which is why i had to adjust/improve the test case logic and also adjust release input and output testdata.

Thanks for the review!